engine/src/main/java/org/terasology/engine/core/bootstrap/EnvironmentSwitchHandler.java

```diff
@@ -116,6 +124,16 @@ public void handleSwitchToGameEnvironment(Context context) {
          * existing then yet.
          */
         unregisterPrefabFormats(assetTypeManager);
+        assetTypeManager.getAssetTypes().forEach(k -> {
+            for (ResourceUrn urn : k.getLoadedAssetUrns()) {
+                if (!urn.getModuleName().equals(TerasologyConstants.ENGINE_MODULE)) {
+                    k.getAsset(urn).ifPresent(Asset::dispose);
+                }
+            }
+        });
```

Comment:
Why do we unregister prefab formats and dispose assets _at this point_?
Shouldn't this be done when exiting the previous world rather than when entering the next? :thinking:

Reply:
Same question as https://github.com/MovingBlocks/Terasology/pull/4829#pullrequestreview-714350912 :upside_down_face:

Reply:
we end up loading those classes again on the classpath so the prefabs are not compatible between sessions.